### ISSUE TITLE
MonitoringNotificationChannel missing displayName field

### ISSUE THREAD

**Issue by sthomson-wyn** [GoogleCloudPlatform/k8s-config-connector#853]
### Checklist

- [X] I did not find a related open enhancement request.
- [X] I understand that enhancement requests filed in the GitHub repository are by default low priority.
- [X] If this request is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

### Describe the feature or resource

MonitoringNotificationChannel is missing displayName field

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/monitoring_notification_channel#display_name

**Comment by diviner524** [comment]
This resource has `displayName` mapped to `name`:

https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/servicemappings/monitoring.yaml#L57

I assume this should give you a displayName same as your K8s resource name. Is that what you have observed?

**Comment by sthomson-wyn** [comment]
Yes, that is the observed behaviour

Though it should be separately configurable, as the display name allows for more options than the RFC 1123 k8s resource name allows for

**Comment by diviner524** [comment]
@sthomson-wyn I agree, it is unfortunate we have introduced this mapping initially, so updating the mapping directly might break some customers who relies on this behavior.

It is possible to re-introduce `displayName` in a way that won't break existing users. Something along the line of if `displayName` is not specified, then use value from the field `metadata.name` instead, although it will require some hacks and considerations.

Or we can remove this mapping when we move the CRD from `v1beta1` to new versions in future.

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
